code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -441,6 +441,25 @@
 	do_sparks(3, 1, src)
 	..()
 
+///Disassembling the bot in a civilized manner with a multitool
+/mob/living/simple_animal/bot/floorbot/disassemble()
+	on = FALSE
+	var/turf/Tsec = get_turf(src)
+	var/obj/item/storage/toolbox/mechanical/N = new /obj/item/storage/toolbox/mechanical(Tsec)
+	N.contents = list()
+	new /obj/item/assembly/prox_sensor(Tsec)
+	drop_part(robot_arm, Tsec)
+	while(amount)// Dumps the tiles into the appropriate sized stacks
+		if(amount >= 16)
+			var/obj/item/stack/tile/plasteel/T = new (Tsec)
+			T.amount = 16
```

Comment:
```suggestion
			var/obj/item/stack/tile/plasteel/T = new (Tsec, 16)
```

Reply:
I changed the code to follow your suggestion in the commit making the floorbots drop the exact amount of floor tile they had left upon disassembly